sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -84,6 +84,26 @@ int pollSocketUntilEventOrTimeout(
   return WSAPoll(&poller, 1, timeout);
 #endif
 }
+
+#ifdef WINDOWS
+// Windows needs this after every write to socket or peformance would be reduced to 1/4 for
+// uploading operation.
+// https://github.com/Azure/azure-sdk-for-cpp/issues/644
+void WinSocketSetBuffSize(curl_socket_t socket)
+{
+  ULONG ideal;
+  DWORD ideallen;
+  // WSAloctl would get the ideal size for the socket buffer.
+  if (WSAIoctl(socket, SIO_IDEAL_SEND_BACKLOG_QUERY, 0, 0, &ideal, sizeof(ideal), &ideallen, 0, 0)
+      == 0)
+  {
+    // if WSAloctl succeded (returned 0), set the socket buffer size.
+    // Specifies the total per-socket buffer space reserved for sends.
+    // https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-setsockopt
+    setsockopt(socket, SOL_SOCKET, SO_SNDBUF, (const char*)&ideal, sizeof(ideal));
```

Comment:
Please add logging around this code.  Capture the return code and log the results.  It will be helpful to have tracing in these code paths.

Reply:
Same as before. This is not a good thing to log because it would flood the log

Reply:
The logging needs to be here and be dialable.  If a customer has a failure they need to be able to turn on the logging to diagnose.  The logging here, like everywhere else, needs to be dialable.